### PR TITLE
Added NED threshold functionality

### PR DESCRIPTION
**Pull request by michael-trollip-AEMO**
Modified response categorisation algorithm, updated changes in shiny, added to manual settings

- [x] pending post processing test

### CODE REVIEW

File: response_categorisation/tests/test_response_categorisation.R [hunk 1]
```diff
@@ -83,7 +86,7 @@ test_that("Test a simple example with one systems through high level function",
   clean <- c("raw", "raw", "raw", "raw", "raw", "raw")
   power_kW <- c(10, 10, 10, 10, 10, 10)
   input_data <- data.frame(c_id, ts, power_kW, d, clean, stringsAsFactors = FALSE)
-  out <- categorise_response(input_data, event_time, window_length)
+  out <- categorise_response(input_data, event_time, window_length,NED_threshold_pct)
```
[the remark this answers is not in the document]
Comment (michael-trollip-AEMO) [reply response_categorisation/tests/test_response_categorisation.R:90]: I've added the NED_threshold_pct variable to each individual test. 
Before making this change the tests worked when I ran it locally as the NED_threshold_pct variable was stored in the global environment. 
Let me know if any issues

File: response_categorisation/tests/test_response_categorisation.R [hunk 2]
```diff
@@ -97,28 +100,111 @@ test_that("Test a simple example with one systems through high level function",
   expect_equal(out, expected_output, tolerance=0.001)
 })
 
-test_that("Test a circuit with a missing pre-event window is 6 Not enough data", {
+test_that("Test a simple example with one systems with no pre event interval", {
```
[the remark this answers is not in the document]
Comment (michael-trollip-AEMO) [reply response_categorisation/tests/test_response_categorisation.R:104]: Here, the pre event interval refers to the one timestamp that occurs immediately before the event (i.e for event at 16:04:46, the 60s pre event interval is at 16:03:55 and the 5s pre event interval is at 16:04:45).

The test is slightly modified from the older test in that it sets the pre event interval as one timestamp ahead. It then includes a timestamp before the pre event interval. It should do everything that the older test does and also account for if there is data before the pre event interval.

File: response_categorisation/tests/test_response_categorisation.R [hunk 3]
```diff
@@ -97,28 +100,111 @@ test_that("Test a simple example with one systems through high level function",
...
+  d <- c(60, 60, 60, 60)
+  c_id <- c("1", "1", "1", "1")
+  clean <- c("raw", "raw", "raw", "raw")
+  power_kW <- c(10, 10, 10, 10)
+  response_category <- c("1 Ride Through", "1 Ride Through", "1 Ride Through", "1 Ride Through")
+  
+  expected_output <- data.frame(c_id, ts, power_kW, d, clean, response_category, stringsAsFactors = FALSE)
+  
+  expect_equal(out, expected_output, tolerance=0.001)
+})
```
[the remark this answers is not in the document]
Comment (michael-trollip-AEMO) [reply response_categorisation/tests/test_response_categorisation.R:159]: Yep that's a good idea, I wrote this by a lot of copy and paste with some minor edits from code that Nick had already written. I won't have time to write it up today but would be happy to do that next week

[the remark this answers is not in the document]
Comment (exfletch) [reply response_categorisation/tests/test_response_categorisation.R:159]: @michael-trollip-AEMO this one test is failing still - looks like it's returning `6 Not enough data` rather that `1 Ride Through`

[the remark this answers is not in the document]
Comment (michael-trollip-AEMO) [reply response_categorisation/tests/test_response_categorisation.R:158]: fixed up, I didnt add a helper function as I realised it was going to take a little more time than expected. Will definitely take that in to consideration for any further testing/development

**Review by exfletch — APPROVED** [review]
Nice one, looks good to merge @michael-trollip-AEMO

**Pull request merged**